### PR TITLE
Fix gift card in transaction

### PR DESCRIPTION
**Pull request by usudaysingh**

### CODE REVIEW

File: gateway.py [hunk 1]
```diff
@@ -125,3 +128,46 @@ def post(cls, transactions):
                 transaction.gift_card.state = 'used'
                 transaction.gift_card.save()
         return rv
+
+
+class PayInvoiceUsingTransactionStart:
+    __name__ = 'account.invoice.pay_using_transaction.start'
+
+    use_gift_card = fields.Boolean("Use Gift Card")
```
Comment (prakashpp) [inline gateway.py]: you dont need this field.

File: gateway.py [hunk 2]
```diff
@@ -125,3 +128,46 @@ def post(cls, transactions):
...
+
+    gift_card = fields.Many2One(
```
Comment (prakashpp) [inline gateway.py]: check if method is gift card: https://github.com/fulfilio/trytond-invoice-payment-gateway/blob/develop/invoice.py#L80

File: gateway.py [hunk 3]
```diff
@@ -125,3 +128,46 @@ def post(cls, transactions):
...
+        'gift_card.gift_card', 'Gift Card', domain=[('state', '=', 'active')],
+        states={
+            'invisible': ~Bool(Eval('use_gift_card')),
```
Comment (prakashpp) [inline gateway.py]: only visible and required when method is gift card

File: gateway.py [hunk 4]
```diff
@@ -125,3 +128,46 @@ def post(cls, transactions):
...
+            'required': Bool(Eval('use_gift_card')),
+        },
+    )
+
+    @staticmethod
+    def default_use_gift_card():
```
Comment (prakashpp) [inline gateway.py]: remove

File: gateway.py [hunk 5]
```diff
@@ -125,3 +128,40 @@ def post(cls, transactions):
...
+            'invisible': Eval('method') != 'gift_card',
+            'required': Eval('method') != 'gift_card'
+        }, depends=['method']
+    )
+
+
+class PayInvoiceUsingTransaction:
+    __name__ = 'account.invoice.pay_using_transaction'
+
+    def create_payment_transaction(self, profile=None):
```
Comment (prakashpp) [inline gateway.py]: implementation of this method is wrong. You are overriding it completely, call super and check if the method is gift card make the changes.

Comment (usudaysingh) [inline gateway.py]: https://github.com/fulfilio/trytond-gift-card/pull/5/files#diff-6d732c0ee74b80254eda6130e2036a81R152

**Pull request closed without merging**